### ISSUE TITLE
C macro version awareness

### ISSUE THREAD

**Issue by g5t** [McStasMcXtrace/McCode#1595]
Version information _in generated C code_ could be useful to component authors wishing to make use of bleeding-edge developments without breaking backwards compatibility for older McCode versions. One way to provide this would be through an integer-valued macro, e.g., `__MCCODE_VERSION__`, created via something like 

```c
#define __MCCODE_VERSION__ 304038L
```

Then a component block that is C code can use, e.g.,
```c
#if defined(__MCCODE_VERSION__) && __MCCODE_VERSION__ >= 304003L
// do something only possible thanks to a patch in 3.4.3
#else
// use a workaround or issue a warning/error
#endif
```

**Comment by g5t** [comment]
This was addressed by #1596

**Comment by g5t** [comment]
This was a duplicate of #1215
